src/field_theory/splitting_field.lean

```diff
@@ -327,6 +327,86 @@ begin
   exact splits_of_splits_id _ h
 end
 
+/--A monic polynomial `p` that has as much roots as its degree can be written `p = ∏(X - a)`, for `a` in `p.roots`-/
+lemma eq_prod_of_card_roots_monic {p : polynomial α}
```

Comment:
Could this proof be made shorter by using `prod_X_sub_C_pow_dvd`?

Reply:
Oh, yes of course! My first idea was to use the remainder and at the end I still used that even with `prod_X_sub_C_pow_dvd`, that makes it completely useless. Now it is better, but I think it can still be improved, for example the part with `tactic.ring_exp.add_pf_sum_z`, where I proved that if `a+b=a` then `b=0` (I didn't find a lemma for that).